bundles/model/org.eclipse.smarthome.model.rule.runtime/src/org/eclipse/smarthome/model/rule/runtime/internal/engine/RuleTriggerManager.java

```diff
@@ -289,11 +325,11 @@ public RuleTriggerManager(Injector injector) {
             case UPDATE:
                 if (newType instanceof State) {
                     State state = (State) newType;
-                    for (Rule rule : rules) {
+                    for (Rule rule : getRulesOrEmptySet(UPDATE, itemName)) {
                         for (EventTrigger t : rule.getEventtrigger()) {
                             if (t instanceof UpdateEventTrigger) {
                                 UpdateEventTrigger ut = (UpdateEventTrigger) t;
-                                if (ut.getItem().equals(item.getName())) {
+                                if (ut.getItem().equals(itemName)) {
```

Comment:
I wonder if it wouldn't make sense to extract this block into a separate method, so it can be shared with the group member code below. Just to reduce the risk of somebody later on changing one and forgetting to adapt the other one...